kukur/source/arrow.py

```diff
@@ -211,3 +223,24 @@ def _map_columns(column_mapping: Dict, data: pa.Table) -> pa.Table:
         columns["quality"] = data[column_mapping["quality"]]
 
     return pa.Table.from_pydict(columns)
+
+
+def _cast_ts_column(
+    data: pa.Table, data_datetime_format: Optional[str], data_timezone: Optional[str]
+) -> pa.Table:
+    if data_datetime_format is None:
+        return data
```

Comment:
It is possible to have timestamps in the data, but in an incorrect time zone. In that case `data_datetime_format` will not be set, but `data_timezone` will. That case will not work because of this early return.

Reply:
But the function `pyarrow.compute.assume_timezone` gives error if it receives aware timestamps. In this case, it will only be possible to use `data_timezone` if there is no timezone in the timestamps.

Reply:
After a while investigating it seems that the **cast** made on the data already converts the timestamps to UTC, according to the schema passed to the function. If timestamps are naive, `data_timezone` can be used. I also added tests for both cases.